specifyweb/frontend/js_src/lib/components/FormParse/fields.ts

```diff
@@ -70,7 +70,13 @@ export type FieldTypes = {
       readonly min: number | string | undefined;
       readonly max: number | string | undefined;
       // These are used by numeric field only
-      readonly step: number | undefined;
+
+      /*
+       * 'any' is not a valid step attribute on the form, and not validated
+       * by the Syncer, but is required for the default step for floating
+       * point fields
+       */
+      readonly step: number | 'any' | undefined;
```

Comment:
The following line was angry about the adjustments to the `Parser` type: 
https://github.com/specify/specify7/blob/e0f1234513aac3e379083524f84da47e539d1f66/specifyweb/frontend/js_src/lib/components/Forms/generateFormDefinition.ts#L272

I have resolved the error by adding the additional `'any'` in the FieldTypes type (which is used for the FormFieldDefinition type). 
Although, I would assume the `FieldTypes` type is supposed to represent the valid values of the attribute for the Cell Definition and `any` is not supported on the form for  the call attribute`initialize="step=value"`

So, if needed, instead of adding the `'any'` step type here, we can alternatively only modify the type definition in the code to autogenerate the form to keep the type definition for the 'Form' elements free of the `Parser` required modifications.